### ISSUE TITLE
Set up wasi-crypto working group?

### ISSUE THREAD

**Issue by rjzak** [WebAssembly/wasi-crypto#73]
I'd like to propose a weekly 30-minute call to discuss wasi-crypto and synchronize efforts. We're ready to use it in [Enarx](https://github.com/enarx/enarx), and would like to have it in a Wasi snapshot.

CC: @npmccallum, @jedisct1, @tarcieri, @sunfishcode 

Refs: #69, #65, #63, #58, #56, https://github.com/WebAssembly/wasi-crypto/pull/71

**Comment by jedisct1** [comment]
Unless this is specifically about rust and wasmtime, @sonder-joker should obviously be here as well.

**Comment by PiotrSikora** [comment]
I'd be interested as well, so please invite me if there is a separate meeting... But is there any reason not to have this discussion (or at least, to kick it off) in the regular bi-weekly WASI meeting?

**Comment by tarcieri** [comment]
I'd be interested in attending a kickoff meeting. Not sure I can attend regularly though. Be happy to jump on to discuss specific issues as they come up

**Comment by rjzak** [comment]
@PiotrSikora Starting off with the Wasi meeting, scheduled for [11 August](https://github.com/WebAssembly/meetings/blob/main/wasi/2022/WASI-08-11.md), works for me.
@jedisct1 Could you tag other potentially interested parties to make them aware?

**Comment by ndevillard** [comment]
Hi all, I would also be interested to join such a working group, if invitations are open?
We are working on a similar project: [a generic C API for crypto.
](https://armmbed.github.io/mbed-crypto/psa/). Hope that makes sense.

**Comment by rjzak** [comment]
Is everyone part of the WebAssembly Working Group? If not, then let's not use the Wasi meeting to get started.

**Comment by jedisct1** [comment]
In order to get some adoption, taking advantage of the crypto extension should be as simple as possible for application developers. Ideally, it should be transparent. Existing crypto libraries and APIs like the generic C API would, under the hood, use the extension when the application is compiled to WebAssembly. This is at least what we're planning to do in the Zig standard library.

So, yes, @ndevillard , your participation would be more than welcome!

**Comment by jedisct1** [comment]
When WebAssembly signatures were discussed during WebAssembly CG meetings, my impression was that most of the participants were not specifically interested in discussing specialized topics such as crypto. Most of the actual discussions actually happened in a dedicated working group.

It may be the same here. A dedicated working group, with its own sessions may be more appropriate, especially to discuss algorithms or implementation details.

**Comment by jedisct1** [comment]
Maybe @autodidaddict @tniessen @vshymanskyy @FiloSottile would like to join?

**Comment by FiloSottile** [comment]
I’d be interested, especially if the goal is for compilation of Go to WASI to automatically use these interfaces. Like Tony, I can definitely attend a kickoff, not sure on all ongoing meetings.

**Comment by linclark** [comment]
> Is everyone part of the WebAssembly Working Group? If not, then let's not use the Wasi meeting to get started.

Unfortunately if this wants to move forward as a WASI proposal, everyone does need to be part of the WebAssembly CG. This is due to IP rules in the W3C.

I would suggest using the WASI SG. We usually have space in the agenda, and other contributors can determine from the agenda whether they are interested in attending or not, so that should solve the "are we boring everyone else" issue.

**Comment by rjzak** [comment]
Joining the WebAssembly Working Group can be done here: https://www.w3.org/community/webassembly/

**Comment by rjzak** [comment]
Thoughts for the first meeting:

* Finalising Wasi-Crypto so it can be in the next Wasi Snapshot.
* Migrate Witx to Wit.
  * Seems Wasmtime Wiggle is the Witx parser, and glue to link the generated code to Wasmtime internals. Alternatives?
* Rust Crypto vs. Boring. Options based on features?
* Anything else?

**Comment by abrown** [comment]
I'm also interested in attending.

**Comment by jedisct1** [comment]
Would the WASI meeting on August 11 (16:00-17:00 UTC) work for everyone?

If not, the next one is on August 25.

**Comment by jedisct1** [comment]
> Seems Wasmtime Wiggle is the Witx parser, and glue to link the generated code to Wasmtime internals. 
> Rust Crypto vs. Boring. Options based on features?

Such discussions may be a bit out of scope, and would be better discussed directly on GitHub (and maybe implementations should have their own repo). They are about inner details of one specific implementation, not about the WASI interface itself.

> Finalising Wasi-Crypto so it can be in the next Wasi Snapshot.

The spec is not finished, but we indeed need to discuss the exact set of required APIs for the first snapshot.

**Comment by rjzak** [comment]
The discussions might be good to have to get people on the same page, at least initially. 

As for the spec being finalised, what's missing? How does it get to be finalised ?

**Comment by npmccallum** [comment]
I'm happy with August 11. I think we should focus on organizing a working group within which we can make design decisions and track progress.

**Comment by jedisct1** [comment]
@FiloSottile @ndevillard @PiotrSikora @sonder-joker would you be able to attend? For the first meeting, your presence would be really useful.

**Comment by PiotrSikora** [comment]
Yes, I'll be there. Thanks for checking!

**Comment by FiloSottile** [comment]
Yeah I can make that, thank you. Do we need to do something (like join the WG) to join?

**Comment by sonder-joker** [comment]
I will attend.

**Comment by jedisct1** [comment]
> Yeah I can make that, thank you. Do we need to do something (like join the WG) to join?

Yes, you need to join [here](https://www.w3.org/community/webassembly/) (same for you, @sonder-joker)

**Comment by FiloSottile** [comment]
@jedisct1 ah, only saw the form now, just filled it, will join if a chair catches it :)

**Comment by npmccallum** [comment]
@jedisct1 Thanks for the talk today!

Any suggestions on day/time for a 30 minute weekly meeting?

**Comment by rjzak** [comment]
I'm sorry I missed the meeting (delayed flight). Are there notes available?

**Comment by rjzak** [comment]
I'm going to get started getting a weekly meeting scheduled next week.

**Comment by jedisct1** [comment]
How about a biweekly meeting at 16:00 UTC on Wednesdays?

**Comment by rjzak** [comment]
I thought maybe the Zulip chat might be easier for communicating than comments on a Github issue, so I created [#wasi-crypto](https://bytecodealliance.zulipchat.com/#narrow/stream/337198-wasi-crypto) and invited some people.

**Comment by rjzak** [comment]
I propose we have a recurring bi-weekly meeting Tuesday at 16:00 UTC, 12:00 Eastern, 09:00 Pacific starting next Tuesday, 30 August. Does that work?
https://meet.google.com/moe-zwuu-hag

**Comment by abrown** [comment]
I believe that conflicts with the WebAssembly CG meeting but if it were pushed a week it would work?

**Comment by rjzak** [comment]
@abrown I have that on my calendar as being on Thursdays. Is that incorrect?

**Comment by abrown** [comment]
I think so? I remember there was some confusion with the calendar invites over the last few months so perhaps one of us has the wrong version. I suspect I have the right one because I have been attending the CG for the last few months without issue.

(...looking this up to see if I can confirm either way)

Yeah, it looks like the CG meetings are on the same Tuesdays as proposed above; see the [CG meeting list](https://github.com/WebAssembly/meetings/tree/main/main/2022), the next of which is on August 30th.

**Comment by rjzak** [comment]
Okay, let's start 6 September. I think I was confused with CG vs. WG.

**Comment by rjzak** [comment]
Could we have these Wasi-Crypto meetings added to the [WebAssembly/meetings](https://github.com/WebAssembly/meetings) page?

**Comment by rjzak** [comment]
@bstrie

**Comment by jedisct1** [comment]
We didn't had meetings for a while due to the host not being present.

@rjzak were you the host? Can you edit the Google calendar entry? Or shall we create a new meeting, possibly with more hosts in order to avoid this issue?

**Comment by rjzak** [comment]
@jedisct1 You (or someone with a more permanent involvement with the project) should re-create the event. The event was in my Google Calendar, which won't last much longer.

**Comment by jedisct1** [comment]
Thank you @rjzak !

New meeting link: https://meet.google.com/yeh-kbzo-pfx
Google calendar: https://calendar.google.com/calendar/event?action=TEMPLATE&tmeid=N3BwZTZuMW02NzI4MTM3MjFkcXVnOThlN25fMjAyMzAzMjFUMTcwMDAwWiBmZGVuaXNAZmFzdGx5LmNvbQ&tmsrc=fdenis%40fastly.com&scp=ALL
